DevelopGitHub/GitHubExtension/GitHubExtension.EntryPoint/FrontEnd/src/client/app/activity/activity.directive.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Created by Vladyslav on 09.04.2016.
+ */
+
+(function () {
+    'use strict';
+
+    angular.module('app.activity')
+        .directive('geActivity', activity);
+
+    activity.$inject = [
+        '$templateCache',
+        '$compile',
+        'logger',
+        'ACTIVITY_CONSTANTS',
+        'i18n'
+    ];
+
+    function activity($templateCache, $compile, logger, constants, i18n) {
+        function link(scope, element) {
+            var type = scope.activity[constants.ACTIVITY_PROPERTIES.TYPE];
```

Comment:
So make one template and move all different parts of the template inside of it (will use ng-switch for selecting right piece) ?